### PR TITLE
Refactors Theme Component and increase coverage

### LINKED ISSUE #1339: Refactor and Cleanup Theme Component
**I'm submitting a ...**
- [ ] bug report
- [ ] feature request
- [x] refactoring and test coverage

**Current behavior:**

- Standard coding practice is not followed
- Code coverage is about 20%, which may be improved

**Expected behavior:**

Resolve above mentioned issues

**Will work on the issue:**
Yes

### PR DESCRIPTION
**Pull request by shreyanshdwivedi**
Fixes #1339 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)
- [x] Added Surge preview link

#### Changes proposed in this pull request:

- Refactored code 
- Added tests to increase coverage

- https://pr-1340-fossasia-susper.surge.sh

### CODE REVIEW

**Comment by shreyanshdwivedi** [comment]
@praveenojha33 @AakashMallik @simsausaurabh we may need to discuss how to tackle the error in travis. This is a well know error which occurs when we test our function. 
Several issues were opened in `angular-cli` org for this, for reference https://github.com/angular/angular-cli/issues/7296

**Comment by praveenojha33** [comment]
@shreyanshdwivedi That issue has been fixed (as mentioned in the comments there). There are some merged PRs also which has referenced. We can look into that and see what changes they have made.
If then also this issue occurs please open an issue there or search/ask on stackoverflow :+1:

**Comment by shreyanshdwivedi** [comment]
@praveenojha33 @simsausaurabh Please review. :)

**Comment by shreyanshdwivedi** [comment]
@simsausaurabh @praveenojha33 please review

**Comment by praveenojha33** [comment]
There are conflicts due to which this PR cannot be merged. 
image Please remove conflicts.

File: src/app/theme/theme.component.spec.ts [hunk 1]
```diff
@@ -28,4 +29,109 @@ describe('ThemeComponent', () => {
...
+
+  it('should set dark theme', async(() => {
+    spyOn(component, 'setTheme').and.callThrough();
+
+    const element = fixture.debugElement.queryAll(By.css('button.theme-link'))[1];
+    const btn = element.nativeElement as HTMLElement;
+    btn.click();
+
+    fixture.detectChanges();
+    fixture.whenStable().then(() => {
```
Comment (sakshee-19) [inline src/app/theme/theme.component.spec.ts:71]: Is it ok to use ```then```? Can't it be done without it?

Comment (shreyanshdwivedi) [reply src/app/theme/theme.component.spec.ts:71]: Actually whatever reference I got about `whenStable` with `async`, I found the same format. I don't think it'll work without this

**Comment by shreyanshdwivedi** [comment]
@praveenojha33 please review now and let me know if the conflicts still exists

**Pull request merged**